libs/sdk-core/src/binding.rs

```diff
@@ -48,16 +48,19 @@ static LOG_INIT: OnceCell<bool> = OnceCell::new();
 /// Wrapper around [BreezServices::connect] which also initializes SDK logging
 pub fn connect(config: Config, seed: Vec<u8>) -> Result<()> {
     block_on(async move {
-        let breez_services =
-            BreezServices::connect(config, seed, Box::new(BindingEventListener {})).await?;
-
-        BREEZ_SERVICES_INSTANCE
-            .set(breez_services)
-            .map_err(|_| SdkError::InitFailed {
-                err: "static node services already set".into(),
-            })?;
-
-        Ok(())
+        let mut locked = BREEZ_SERVICES_INSTANCE.lock().await;
+        match *locked {
+            None => {
+                let breez_services =
+                    BreezServices::connect(config, seed, Box::new(BindingEventListener {})).await?;
```

Comment:
The connect is being called after acquiring the lock but disconnect is not. Better to be consistent here.

Reply:
Done. I changed `disconnect` to also first acquire the lock.